librz/core/basefind.c

```diff
@@ -276,12 +318,12 @@ static void *basefind_thread_ui(BaseFindUIInfo *ui_info) {
 			BaseFindThreadData *bftd = rz_th_get_user(th);
 			basefind_set_thread_info(bftd, &th_info, i);
 			if (!callback(&th_info, user)) {
-				rz_th_pool_kill(pool);
+				basefind_stop_all_search_threads(pool);
 				goto end;
 			}
 		}
 		rz_sys_usleep(100000);
-	} while (1);
+	} while (thread_bool_get(&ui_info->loop));
```

Comment:
is there anything preventing this condition to be true even though, one microsecond after this has been checked, the value of ->loop is set to false? Is it alright to have such case?

Reply:
What may happen is that while exiting from the `thread_bool_get` which holds a `true`, the value gets set to `false` but this means that the execution is still happening, which is fine, since it will be terminated during the next loop.
The thread is anyway waited by the main thread, so nothing to worry.